staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1017,11 +1017,8 @@ func (e *Store) DeleteCollection(ctx context.Context, deleteValidation rest.Vali
 	// DeleteCollectionWorkers (it doesn't make much sense to spawn 16
 	// workers to delete 10 items).
 	workersNumber := e.DeleteCollectionWorkers
-	if workersNumber > len(items) {
-		workersNumber = len(items)
-	}
-	if workersNumber < 1 {
-		workersNumber = 1
+	if((workersNumber < 1) || (workersNumber > len(items)){
+		return nil, err
```

Comment:
That change isn't correct.

The workersNumber > len(item) is perfectly valid situation (e.g. when we delete just a couple items, or potentially even zero items).